### PR TITLE
chore: part 3 of replacing BigtableOptions to BigtableHBaseSettings

### PR DESCRIPTION
**Pull request by rahulKQL**
This change updates `AbstractBigtableRegionLocator` and `AbstractBigtableTable` and it's references.

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/AbstractBigtableRegionLocator.java [hunk 1]
```diff
@@ -56,11 +57,16 @@
...
 
   public AbstractBigtableRegionLocator(
-      TableName tableName, BigtableOptions options, IBigtableDataClient client) {
+      TableName tableName, BigtableHBaseSettings settings, IBigtableDataClient client) {
     this.tableName = tableName;
     this.client = client;
-    this.bigtableTableName = options.getInstanceName().toTableName(tableName.getNameAsString());
-    ServerName serverName = ServerName.valueOf(options.getDataHost(), options.getPort(), 0);
+    this.bigtableTableName =
+        new BigtableTableName(
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/AbstractBigtableRegionLocator.java]: how frequently do we need to do this operation? Is it worth adding this as a getter on `BigtableHbaseSettings` (similar to the convenience method `toTableName()`)?

Comment (rahulKQL) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/AbstractBigtableRegionLocator.java]: This is an intermediate state, We would be able to replace `BigtableTableName` with string `tableID` once the veneer client wrappers are in place.

Comment (kolea2) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/AbstractBigtableRegionLocator.java]: sounds good, thanks!

**Comment by rahulKQL** [comment]
rebased and added a unit to fix CodeCov build, which had -0.02% coverage change.

**Pull request merged**